RA - Add Soviet08a

Smittytron opened the pull request:
This is a standard destroy the enemy mission. @Murto did the early triggers and I picked up where he left off.

Muh-Muh commented:
Small issue:
If you capture the enemies production facilities some things can not be built (bc. tech centre can't be built) and should be disabled. These are:
Gap generator
Chronosphere
Tanya (x2, I assume the second Tanya is the .autotarget one)
Radar Jammer
Longbow
Cruiser

Other than that I didn't find any issues playing it (so far only on easy) or in the code.

Smittytron commented:
Updated

Muh-Muh commented:
Is it possible that you forgot to push? rules.yaml still seems to be missing those actors.

Smittytron commented:
I didn't put them at the end, I grouped them with their type i.e. structures, vehicles ect.
MRJ, HELI, CA, GAP, PDOX and E7 are all in there.

Muh-Muh commented:
Oh right. Nevermind then.

abcdefg30 commented on mods/ra/maps/soviet-08a/soviet08a.lua:
```diff
@@ -0,0 +1,160 @@
+end
+
+Tick = function()
+	greece.Cash = 1000
```
This feels like we're hiding some other issue. What exactly was the reason again why the AI to runs out of money? Do the harvesters get stuck?

Smittytron replied:
I just figured there's no way AI can keep up constant infantry and vehicle production off of one harvester mining long-distance. That takes four in competitive play.

The mission Evacuation does Cash = 1000 as well.

MunWolf replied:
I really don't like cheating AI, but I guess if we didn't this mission would be too easy :/

Mailaender replied:
The map doesn't seem to have ore mines so the trucks run low soon unless you expand.

abcdefg30 commented on mods/ra/maps/soviet-08a/rules.yaml:
```diff
@@ -0,0 +1,101 @@
+	RevealsShroud:
+		Range: 6c0
+
+LST.Reinforcement:
```
Those should be unselectable or have no voice when selecting them.

abcdefg30 reviewed (commented):
Some of the sandbag walls probably have the wrong owner, as they are crushed by allied tanks.

Smittytron commented on mods/ra/maps/soviet-08a/rules.yaml:
```diff
@@ -0,0 +1,101 @@
+	RenderSprites:
+		Image: lst
+
+POWERPROXY:
```
This one is the 1:30 paradrop that gives three flamers. The reinforcement and retreat triggers (sans the mammoth) were written by Murto and are apparently true to the original.

Smittytron replied:
Renamed to `scripteddrop` for clarity.

Smittytron commented:
Updated

MustaphaTR commented:
Mechanic and Helipad are available if you capture the allied tech.

Smittytron commented:
Updated

GraionDilach commented on mods/ra/maps/soviet-08a/soviet08a-AI.lua:
```diff
@@ -0,0 +1,131 @@
+AlliedVehicleType = "Normal"
+AlliedVehicles =
+{
+	Normal = { "1tnk", "2tnk" },
```
Wouldn't it make sense to produce jeeps as well at this point?

Smittytron committed:
Add Soviet08a

Smittytron commented:
Rebased

abcdefg30 commented on mods/ra/maps/soviet-08a/soviet08a.lua:
```diff
@@ -0,0 +1,161 @@
+		end)
+	end)
+
+	Harvester.FindResources()
```
This loos odd. Doesn't the harvester start mining on its own?

Smittytron replied:
The refinery spawned one will. This is to get the one at 57,56 going.

Smittytron committed:
Add myself to AUTHORS

Smittytron commented:
Updated: Tested this without bounties and saw that I all of a sudden couldn't make it off the beachhead on highest difficulty. Rebalanced to the point where I could beat it again with an AI delay and added a money crate to the church.

reaperrr commented:
Moving to Next+1 milestone so it's position in the PR queue won't make it miss another release.

Orb370 reviewed (dismissed):
Played through the mission on normal difficulty. No issues.

pchote commented:
@Smittytron are you able to look into the issue that @SoScared raised? Do we have any other blockers for getting this merged (aside from needing a rebase)?

Smittytron commented:
> @Smittytron are you able to look into the issue that @SoScared raised? Do we have any other blockers for getting this merged (aside from needing a rebase)?

I don't think there is a way to fix SoScared's point. I assume the transport would get through after the subs move out of the way.

The only potential blocker is @abcdefg30 asked me to look at adding Allied naval attacks to this mission. I've tried both setting up production and having ships come in off the map, but so far I haven't been able to get either method to work. I'll have more time to try to figure it out around the Thanksgiving holiday.

Smittytron commented:
Updated.
Added two naval patrols. I also removed Real tough guy difficulty and added a gem mine by the allied base.
I added the mine to a) Give the allied harvester more to do, and b) to ensure that if the player has mined the map out after destroying the allied base, they will have a source of income to finish the naval units off with.
